### ISSUE TITLE
attachment.data returns the oldest content, not the latest

### ISSUE THREAD

**Issue by sergiofagostinho** [libkeepass/pykeepass#143]
Hi,

We've found the following code:
```python
...
for entry in entries:
  ...
     attachment = entry.attachments[0]
     secret = attachment.data
```

only works when you delete the history.

Doesn't matter if you delete the attachment or add it with a different name.

I've attached a test file (password: `password`) where I've added a PNG file, deleted it and added a text file. If you run the code it will still display the binary data.

kbdx file: https://drive.google.com/open?id=1ms3Jobn_rxfDTrmK2bE2x6Mn3OZzxIMK

Also, I'm writing this file with the latest version of KeePassX on Mac (2.0.3).

**Comment by Evidlo** [comment]
Could you provide a more complete example of what you mean?  Everything seems to work as expected for me.

**Comment by sergiofagostinho** [comment]
Hi @Evidlo. I would expect the `attachment.data` to hold the current file content (text), not the file I've deleted (PNG image). Basically, what I see in Keepass X:

![image](https://user-images.githubusercontent.com/2901192/54019004-0069d300-4182-11e9-880c-ff70cde1fe81.png)

**Comment by pscosta** [comment]
I'm able to reproduce the issue calling the following method:

`find_attachments (id=None, filename=None, element=None, recursive=True, regex=False, flags=None, history=False, first=False)`

even thought I set `history=False` I always get old  attachments from the history.
The workaround for me is to delete all history entries before retrieving attachments.

**Comment by Evidlo** [comment]
Fixed by PR #147
